common/datatype_document.cc

```diff
@@ -483,7 +504,9 @@ datatype_document :: coerce_binary_to_primitive(const e::slice& in,
     else
     {
         *type = HYPERDATATYPE_DOCUMENT;
-        *value = in;
+        scratch->resize(in.size());
+        memmove(scratch->data(), in.data(), in.size());
+        *value = e::slice(&(*scratch)[0], in.size());
```

Comment:
I assume this was a bug? Slice was copied but the data it points to might be destroyed.